engine/src/cmd/music.cpp

```diff
@@ -514,7 +514,7 @@ void Music::GotoSong(int whichlist, int whichsong, bool skip, int layer) {
         if (whichsong != NOLIST && whichlist != NOLIST && whichlist < (int) playlist.size() && whichsong
                 < (int) playlist[whichlist].size()) {
             if (muzak[(layer >= 0) ? layer : 0].lastlist != whichlist) {
-                if (configuration()->audio_config.shuffle_songs.clear_history_on_list_change) {
+                if (configuration()->audio.shuffle_songs_section.clear_history_on_list_change) {
```

Comment:
If you're going to remove the config suffix, why not the section suffix as well.
Brevity is the soul of wit.

Reply:
There was another setting simply called `audio.shuffle_songs`. To distinguish between that single setting and this setting section, I added the `_section` at the end.